### PR TITLE
feat: adding cart functionality to e-commerce

### LINKED ISSUE #21: adding cart functionality
adding cart functionality to e-commerce

### PR DESCRIPTION
**Pull request by mariyasada**
- adding show message when a product doesn't match with user criteria after applying filter
- adding cart message (your cart is empty)
- creating the same context for wishlist and cart
- adding wishlist item reducer ad cart reducer
- creating separate initial-state for wishlist and cart for now because it makes me confused
- adding Add to cart and increment and decrement function using reducer

https://user-images.githubusercontent.com/77275100/159112034-32bf0403-6357-4995-ad84-b6063f642f4a.mp4

r

### CODE REVIEW

File: src/context/Reducer/WishListItemReducer.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const WishListItemReducer=(state,action)=>{
+    const {wishListItem,wishListCount}=state;
+    const {payload}=action;
```
Comment (JantuDeb) [inline src/context/Reducer/WishListItemReducer.js:3]: You can destructure `type`here

**Review by JantuDeb — APPROVED** [review]
If you  deleted whole lot of codes then don't put it in one commit, it is very difficult to read the codes.

File: src/context/Reducer/WishListItemReducer.js (hunk 1 above)
Comment (mariyasada) [reply src/context/Reducer/WishListItemReducer.js:3]: Will do. But what happened I forgot to commit some changes before ,that is why this file and commits are too big

File: src/components/ProductCard/Productcard.jsx [hunk 2]
```diff
@@ -1,59 +1,93 @@
...
-                        </Link>                      
-                    </span>
-                  </div>
-                </div>
-              </div>
-  )
-}
-
+      <div className="card-image-holder">
+        <Link to="">
```
Comment (Bhakti199) [inline src/components/ProductCard/Productcard.jsx]: add `"#"` here

**Review by Bhakti199 — APPROVED** [review]
This code looks fine to me, nicely written and structured.
proper use of useReducer method for cart and wishlist management.
This PR is good to merge.

File: src/components/ProductCard/Productcard.jsx (hunk 2 above)
Comment (mariyasada) [reply src/components/ProductCard/Productcard.jsx]: will do

**Pull request merged**